src/types/BulkUpload.ts

```diff
@@ -0,0 +1,41 @@
+import { ajv } from '../ajv';
+import type { JSONSchemaType } from 'ajv';
+
+export enum BulkUploadStatus {
+  PENDING = 'pending',
+  IN_PROGRESS = 'in_progress',
+  COMPLETED = 'completed',
+  FAILED = 'failed',
+  CANCELED = 'canceled',
+}
+
+export interface BulkUpload {
+  readonly id: number;
+  fileName: string
+  sourceUrl: string;
+  status: BulkUploadStatus;
+  readonly createdAt: Date;
+}
+
+// See https://github.com/typescript-eslint/typescript-eslint/issues/1824
+/* eslint-disable @typescript-eslint/indent */
+export type BulkUploadCreate = Omit<BulkUpload, 'createdAt' | 'status' | 'id'>;
+/* eslint-enable @typescript-eslint/indent */
+
+export const bulkUploadCreateSchema: JSONSchemaType<BulkUploadCreate> = {
+  type: 'object',
+  properties: {
+    fileName: {
+      type: 'string',
+    },
+    sourceUrl: {
+      type: 'string',
+    },
```

Comment:
Should we specify some minimum lengths here?

```suggestion
    fileName: {
      type: 'string',
      minLength: 1,
    },
    sourceUrl: {
      type: 'string',
      minLength: 1,
    },
```

`1` is obviously way too short, but I don't know how in-the-weeds you want to get here; for the file name, `5` plus "ends in `.csv`" might be more reasonable, and we've previously discussed checking that the URL comes from our S3 bucket.

Reply:
Good call -- will add both of those in.

Reply:
Update on the "check from our S3 bucket" concept -- I'm going to hold off on adding that to the specification itself (in my head I'm leaving it as "that is less about the structure / schema of the data but rather is a more dynamic type of validation" but maybe that's not a useful distinction.)

I may just say "it has to start with `https?:\/\/`" for now.